src/filter/exceptions.filter.ts

```diff
@@ -0,0 +1,27 @@
+import { ExceptionFilter, Catch, ArgumentsHost, HttpException, UnauthorizedException, BadRequestException } from '@nestjs/common';
+import { Response } from 'express';
+import { ErrorResponse } from '../dto/error-response.dto';
+
+@Catch()
+export class ExceptionsFilter implements ExceptionFilter {
+  catch(exception: any, host: ArgumentsHost) {
+    const ctx = host.switchToHttp();
+    const response = ctx.getResponse<Response>();
+    let status = exception instanceof HttpException ? exception.getStatus() : 500;
+
+    const errorResponse = new ErrorResponse();
+    errorResponse.message = exception.message
+
+    if (exception instanceof UnauthorizedException) {
+        status = 401;
+    }
+    if (exception instanceof BadRequestException) {
```

Comment:
Isn't there always status and message information in HttpException? This check seems unnecessary